premailer/premailer.py

```diff
@@ -306,6 +308,16 @@ def transform(self, html=None, pretty_print=True, **kwargs):
             else:
                 parser = etree.HTMLParser()
             stripped = html.strip()
+
+            if self.preserve_handlebar_syntax:
+                stripped = re.sub(
+                    r'="{{(.*?)}}"',
+                    lambda match: '="{{'
+                    + match.groups()[0].replace('"', "%22")
+                    + '}}"',
+                    stripped,
+                )
```

Comment:
This needs to be explained. Because it's quite surprising. 

- One explanation that explains that Handlebar syntax might have its own characters that would normally get double encoded by `etree.tostring()`. Perhaps include an example from your unit test in the comment. (remember to not explain what the code does)
- One explanation for why this can't be done in the post-processing step on line 533.

Regarding the second one, I'm actually curious why it can't be done as part of that `unquote(...)` stuff you're doing around line 533.

Reply:
Regarding these explanations: You just want me to leave comments in the code to explain why the implementation was necessary and why a part of the implementation needed to occur prior to being encoded, right?

Point of clarity: The reason I thought I would have to escape the `"` characters prior to being encoded was that `etree.tostring()` looks at an attribute and treats its contents from the beginning `"` to the very next `"`. That's why:
```html
<a href="{{ data | default: "Test & <code>" }}"></a>
```
turns into
```html
<a href="%7B%7B%20data%20%7C%20default:%20" test>" }}"&gt;</a>
```
If we had to wait until after the encoding, it would seem that getting the original value back would be very difficult due to how `etree.tostring()` encodes attributes (as seen above).

Reply:
> Regarding these explanations: You just want me to leave comments in the code to explain why the implementation was necessary and why a part of the implementation needed to occur prior to being encoded, right?

Exactly. 

I usually think, "Does this code make me have to stop and think? If not today, what about when I (or someone else!) read this code in 6 months."

Reply:
Working on this now

Reply:
I might have been a little too verbose with my comments but I'd rather do that than have people not understand what's going on: https://github.com/peterbe/premailer/pull/252/commits/8426f81e1e8200ac7ac80584036ea3f8c46fa799 Let me know if you want me to change anything.